src/vc_sqnm/historylist.hpp

```diff
@@ -0,0 +1,66 @@
+/**
+ * @file historylist.hpp
+ * @author Moritz Gubler (moritz.gubler@unibas.ch)
+ * @brief Implementation of the SQNM method. More informations about the algorithm can be found here: https://aip.scitation.org/doi/10.1063/1.4905665
+ * @date 2022-07-13
+ * 
+ */
+
+#ifndef HISTORYLIST_HPP
+#define HISTORYLIST_HPP
+#include <Eigen/Dense>
+
+namespace hlist_space{
+  class HistoryList {
+    private:
+    int i_count = 0;
+    int nhistx;
+    int ndim;
+    Eigen::VectorXd oldElem;
+
+    public: 
+
+    Eigen::MatrixXd hlist;
+    Eigen::MatrixXd difflist;
+    Eigen::MatrixXd normalized_difflist;
+
+
+    HistoryList(int &ndim_, int &nhistx_)
+    {
+      nhistx = nhistx_;
+      ndim = ndim_;
+      hlist.resize(ndim, nhistx);
+      difflist.resize(ndim, nhistx);
+      normalized_difflist.resize(ndim, nhistx);
+      i_count = 0;
+    }
+
+    int add(Eigen::VectorXd &x_in)
```

Comment:
I don't see that you're modifying `x_in`, so you should pass by `const` reference.
```suggestion
    int add(const Eigen::VectorXd &x_in)
```

Reply:
Ok, I did that [here](https://github.com/MRChemSoft/mrchem/pull/451/commits/c822f09c2bd99c8a81b532dfed58820fa97beec8)